Source/AliveLib/Slig.cpp

```diff
@@ -6592,11 +6581,14 @@ BaseAliveGameObject* Slig::FindBeatTarget_4BD070(int /*a2*/, int gridBlocks)
             PSX_RECT bRect = {};
             pObj->vGetBoundingRect_424FD0(&bRect, 1);
 
-            if (pObj->field_10C_health > FP_FromInteger(0) &&
+            if (hitRect.w <= bRect.w &&
+                hitRect.x >= bRect.x &&
+                hitRect.y >= bRect.y &&
+                hitRect.h <= bRect.h &&
                 pObj->field_D6_scale == field_D6_scale &&
                 !IsInInvisibleZone_425710(pObj) &&
                 !pObj->field_114_flags.Get(Flags_114::e114_Bit8_bInvisible) &&
-                PSX_Rects_overlap_no_adjustment(&hitRect, &bRect))
+                pObj->field_10C_health > FP_FromInteger(0))
```

Comment:
didn't this make more sense to be first? Do the cheap calc before the expensive ones?

Reply:
It does, however we're not at the refactoring stage yet. All I'm doing here is make it IDA compliant